### PR TITLE
Vault interaction events, part 1

### PR DESCRIPTION
**Pull request by levity**
Add to the `interaction_events` table for vault actions (create, deposit, withdraw, distribute, claim).

Also refactor vault action handlers so their dependencies are more co-located.

- [x] create
- [x] deposit
- [x] withdraw
- [x] distribute

### CODE REVIEW

File: api-lib/handlerHelpers.ts
Comment (levity) [inline api-lib/handlerHelpers.ts:10]: @topocount what do you think of this lil guy?

Comment (levity) [reply api-lib/handlerHelpers.ts:10]: @crabsinger this is an attempt at reducing action boilerplate. i first started with the idea of wrapping the entire handler function and passing these parsed props into it, but getting typing working was surprisingly difficult

**Comment by levity** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1378** <a href="https://app.graphite.dev/github/pr/coordinape/coordinape/1378" target="_blank">Graphite</a> 
    * **PR #1376** <a href="https://app.graphite.dev/github/pr/coordinape/coordinape/1376" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/coordinape/coordinape/1376?utm_source=stack-comment).

**Comment by levity** [comment]
Opened this up for review even though it just supports tracking create, deposit, & withdraw. Will do the rest in later PRs

**Comment by levity** [comment]
The test failure was a legit issue; fixed now, but will leave this as a draft until I can clean up the diff & move some tangential fixes to other PRs

**Review by crabsinger — DISMISSED** [review]

**Comment by levity** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/coordinape/coordinape/1376?utm_source=stack-comment).


Job Reference: Uypn65k7WAiFONVxIiti

**Pull request merged**
